libraries/AP_Eigen/AP_MatrixAddons.h

```diff
@@ -0,0 +1,136 @@
+/*
+ * This is an extension file for the Eigen library
+ * @authors: Daniel Frenzel <dgdanielf@gmail.com>
+ */
+#include <AP_Eigen/AP_Algebra.h>
+
+
+/*
+ * @brief If <T> is a floating point type, 
+ * this functions rounds to the closest integer and returns it as <T>.
+ */
+template <class IntType, class FloatType>
+EIGEN_STRONG_INLINE static Matrix<IntType, _Rows, _Cols, _Options, _MaxRows, _MaxCols> get_nearest_integral(const Matrix<FloatType, _Rows, _Cols, _Options, _MaxRows, _MaxCols> &other) {
+    if(!std::is_integral<IntType>::value || !std::is_floating_point<FloatType>::value) {
+        printf("Matrix::get_nearest_integral(): template parameters do not make sense\n");
+    }
```

Comment:
Actually, I was also in doubt here. I wanted to give a message that the function is not used as intended, but a compiler error was in my eyes too rude. On the other hand, a printf is slow. For performance reasons I will replace it.